app/Http/Controllers/UserController.php

```
@@ -0,0 +1,58 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Session;
+use Illuminate\Support\Facades\Hash;
+use App\Models\UserModel as User;
+
+final class UserController 
+{
+    public function signup(Request $request)
+    {
+        $validatedData = $request->validate([
+            'username' => 'required|unique:users|max:100',
+            'email'    => 'required|email|max:250',
+            'password' => 'required|min:8|max:60',
+        ]);
+
+        $user = new User();
+        $user->username = $validatedData['username'];
+        $user->email = $validatedData['email'];
+        $user->password = Hash::make($validatedData['password']);
+        $user->status = 0;
+
+        $user->save();
+
+        session($validatedData);
+
+        return back()->withErrors('');
```

Maybe this line right here is the reason why it is giving `silent errors`

---

errors are blank for now for test implementation, Signup/signin will eventually converted to APIs